### ISSUE TITLE
Upgrading to 1.7.1 Breakes application 

### ISSUE THREAD

**Issue by alecslupu** [alexreisner/geocoder#1557]
### Expected behavior
Seamless upgrade from 1.7.0 to 1.7.1

### Actual behavior
Exception is raised: 
```
NameError: uninitialized constant Geocoder::CacheStore::Base
/bundle_cache/ruby/2.7.0/gems/geocoder-1.7.1/lib/geocoder/cache_stores/redis.rb:2:in `<module:CacheStore>'
/bundle_cache/ruby/2.7.0/gems/geocoder-1.7.1/lib/geocoder/cache_stores/redis.rb:1:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/bundle_cache/ruby/2.7.0/gems/geocoder-1.7.1/lib/geocoder/cache.rb:1:in `block in <main>'
/bundle_cache/ruby/2.7.0/gems/geocoder-1.7.1/lib/geocoder/cache.rb:1:in `each'
/bundle_cache/ruby/2.7.0/gems/geocoder-1.7.1/lib/geocoder/cache.rb:1:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/bundle_cache/ruby/2.7.0/gems/geocoder-1.7.1/lib/geocoder.rb:7:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/code/decidim/decidim-core/lib/decidim/core/engine.rb:28:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/code/decidim/decidim-core/lib/decidim/core.rb:3:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/code/decidim/lib/decidim.rb:3:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/usr/local/bundle/gems/bundler-2.2.22/lib/bundler/runtime.rb:66:in `block (2 levels) in require'
/usr/local/bundle/gems/bundler-2.2.22/lib/bundler/runtime.rb:61:in `each'
/usr/local/bundle/gems/bundler-2.2.22/lib/bundler/runtime.rb:61:in `block in require'
/usr/local/bundle/gems/bundler-2.2.22/lib/bundler/runtime.rb:50:in `each'
/usr/local/bundle/gems/bundler-2.2.22/lib/bundler/runtime.rb:50:in `require'
/usr/local/bundle/gems/bundler-2.2.22/lib/bundler.rb:174:in `require'
/code/config/application.rb:7:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:54:in `require_relative'
/code/Rakefile:4:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:60:in `load'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:60:in `load'
/bundle_cache/ruby/2.7.0/gems/railties-6.0.4.4/lib/rails/commands/rake/rake_command.rb:22:in `block in perform'
/bundle_cache/ruby/2.7.0/gems/railties-6.0.4.4/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/bundle_cache/ruby/2.7.0/gems/railties-6.0.4.4/lib/rails/command.rb:48:in `invoke'
/bundle_cache/ruby/2.7.0/gems/railties-6.0.4.4/lib/rails/commands.rb:18:in `<main>'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/bundle_cache/ruby/2.7.0/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
bin/rails:4:in `<main>'
(See full trace by running task with --trace)
```

### Steps to reproduce
I am running a fresh install on a rails 6.0.4, with Geocoder, no configuration (like cache store) done on Geocoder

### Environment info

* Geocoder version: 1.7.1
* Rails version:  6.0.4.4
* Ruby version: 3.0.2
* Database (if applicable):  
* Lookup (if applicable):

**Comment by alexreisner** [comment]
@alecslupu Thanks for reporting this. Oddly, I'm having a hard time reproducing it. Would you mind trying something to see if it fixes the problem? In your Gemfile, replace the current `gem 'geocoder'` line with this:

    gem 'geocoder', github: 'alexreisner/geocoder', branch: 'require-cache-store-base'

and re-run `bundle install`. Let me know if that changes anything.

(cc @shqear93)

**Comment by stuartrr** [comment]
I have the same issue on one machine but none of my other machines (all amazon linux systems but some differences in setups). Rails 5.2.6. I can confirm that branch `require-cache-store-base` fixes it.

**Comment by alecslupu** [comment]
@alexreisner, strangely enough, i cannot reproduce the initial issue so i can test if the proposed fix works or not. 

Maybe the confirmation from @stuartrr it is enough for a 1.7.2?

**Comment by alexreisner** [comment]
@stuartrr thanks for confirming! Just released 1.7.2.

**Comment by shqear93** [comment]
@alecslupu Thanks for reporting this!

For some reason I couldn't reproduce it, and for some reason it happens when running tests only (`rspec`) and only on GitHub actions pipeline and not locally! thanks for resolving this guys 🥷
